### PR TITLE
Next

### PR DESCRIPTION
**Pull request by sunil-lakshman**
Ruby - Azure-EU and include_metadata Implementation in Ruby Delivery SDK

Adding Region Support for Azure-EU

Adding support for include_metadata in Asset, Entry and Query classes.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,35 +1,37 @@
 PATH
   remote: .
   specs:
-    contentstack (0.6.3)
-      activesupport (~> 3.2)
+    contentstack (0.6.4)
+      activesupport (>= 3.2)
       contentstack_utils (~> 1.0)
```
Comment (ishaileshmishra) [inline Gemfile.lock:6]: Do we need to update the Util package to 1.1.3.2 ?
https://rubygems.org/gems/contentstack_utils/versions/1.1.3.2

File: lib/contentstack/client.rb [hunk 2]
```diff
@@ -80,13 +81,43 @@ def sync(params)
 
     private
     def get_default_region_hosts(region='us')
+      host = "https://cdn.contentstack.io" #set default host if region is nil
```
Comment (ishaileshmishra) [inline lib/contentstack/client.rb:84]: Can you separate https protocol from the host string. Make a const variable const string PROTOCOL = "https://" and use it wherever it is required

File: lib/contentstack/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module Contentstack
-  VERSION = "0.6.3"
+  VERSION = "0.6.4"
```
Comment (ishaileshmishra) [inline lib/contentstack/version.rb:2]: Check versioning again. If we are adding new function then it should go for v0.7.0

File: spec/entry_spec.rb [hunk 4]
```diff
@@ -87,6 +87,11 @@
     data = category.include_content_type.fetch
     expect(data.content_type).not_to be nil
   end
+
+  it "should get data using `include_metadata` method" do
```
Comment (ishaileshmishra) [inline spec/entry_spec.rb:91]: Add two testcases: With flag tru and false in include_metdata

**Review by ishaileshmishra — COMMENTED** [review]
You have added 2 feature in the code
- include_metadata 
- Region support for Azure-EU

Test cases added partially for the include_metadata. and there are no testcase added for the region support.

File: lib/contentstack/region.rb [hunk 5]
```diff
@@ -2,5 +2,10 @@ module Contentstack
   class Region
     EU='eu'
     US='us'
+    AZURE_NA='azure-na'
+    AZURE_EU='azure-eu'
+    PROTOCOL='https'
```
Comment (ishaileshmishra) [inline lib/contentstack/region.rb:7]: I think PROTOCOL = 'https://', In this case you do not have to string concate :// while creating endpoint

**Review by ishaileshmishra — COMMENTED** [review]
1 change : Use Protocol = 'https://'

File: lib/contentstack/region.rb (hunk 5 above)
Comment (uttamukkoji) [reply lib/contentstack/region.rb:7]: @sunil-lakshman PROTOCOL should no be in Region. Region is basically DB region.

File: lib/contentstack/region.rb [hunk 6]
```diff
@@ -2,5 +2,10 @@ module Contentstack
   class Region
     EU='eu'
     US='us'
+    AZURE_NA='azure-na'
+    AZURE_EU='azure-eu'
+    PROTOCOL='https://'
```
Comment (uttamukkoji) [inline lib/contentstack/region.rb:7]: @sunil-lakshman Please make different class for constant Region should no contain the constants other than DB regions.

**Review by ishaileshmishra — APPROVED** [review]
approved

**Review by aravindbuilt — APPROVED** [review]

**Pull request merged**